packages/autorest.python/autorest/codegen/templates/vendor.py.jinja2

```diff
@@ -66,16 +66,31 @@ def prep_if_none_match(etag: Optional[str], match_condition: Optional[MatchCondi
     return None
 {% endif %}
 {% if code_model.has_form_data %}
+
+# file-like tuple could be `(filename, IO (or bytes))` or `(filename, IO (or bytes), content_type)`
+FileType = Union[IOBase, bytes]
+FileTuple = Union[Tuple[str, FileType], Tuple[str, FileType, str]]
+MultiPartFile = Union[IOBase, bytes, FileTuple]
+HandledMultiPartFile = Union[IOBase, FileTuple]
+
 class NamedBytesIO(BytesIO):
     def __init__(self, name: str, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.name = name
 
-def multipart_file(file: Union[IOBase, bytes]) -> IOBase:
-    if isinstance(file, IOBase):
+def has_file(data: Any) -> bool:
```

Comment:
I don't think we should do this much checking on the file body, honestly we maybe should just pass the value straight to core cc @lmazuel

The only thing we should watch out for is that we concatenate multiple files. I.e., if we get
`def func(profile_image: IO, pictures: IO[])`, then we should make sure we send `files=profile_image + pictures`

Reply:
Make sense. Codegen knows which part of the body is file type so we don't need to judge it in runtime.
Another thing to confirm is whether Python SDK permit users input bytes directly?